lavague-core/lavague/core/navigation.py

```diff
@@ -271,7 +294,11 @@ def execute_instruction_gradio(self, instruction: str, action_engine: Any):
                     context_str=llm_context, query_str=instruction
                 )
                 response = self.llm.complete(prompt).text
-                action = self.extractor.extract(response)
+                try:
+                    action = self.extractor.extract(response,action_shape_validator=self.action_shape_validator)
+                except Exception as e:
+                    logging_print.error(f"Error extracting action: {e}")
+                    break
```

Comment:
Instead of `break` we can continue so the LLM gets another chance to generate valid action until we reach `n_attempts` limit

```suggestion
                    continue
```